src/smbprotocol/connection.py

```diff
@@ -947,6 +950,8 @@ def disconnect(self, close=True, timeout=None):
 
         log.info("Disconnecting transport connection")
         self.transport.close()
+        if self._t_worker:
+            self._t_worker.join(timeout=2)
```

Comment:
Have you noticed this timeout being hit in real life? If it is timing out it might be worth investigating why the thread is still running.

Reply:
I have never experienced the timeout.
This is why I didn't bother to do a check for timeout.

The thread stops... it's just that when running the automated tests, I am askinging for the list of active threads, just before the thread stop.

also, this is really a problem only during automated testing.